### PR TITLE
ROX-20291: Move FE sorting of Policies table to avoid re-render

### PR DESCRIPTION
**Pull request by vjwilson**
## Description

This PR refactors the client-side sorting on the Policies table, by moving it up from the `PoliciesTable.tsx` component file, up to its parent `PoliciesTablePage.tsx` component. This is done to prevent the layout shift that occurs when the table was being rendered with an empty array, and then fed an unsorted API response, and then immediately re-rendered with the sorted list, which was sorted in a useEffect block. The initial render did not have wide empty header for the turndown triangle column--PatternFly did not render that header if there were no rows--and the subsequent layout shift was causing frequent "element is animating" failures in CI.

I decided move the sorting up into the promise-resolved block for the `/policies` API call in the parent, because that API call gets refreshed on Import, as well as Deletion and Reassess All functions. I could have attempted a more radical refactoring to move the API call down, but this seemed like a safer, easier-to-review option.

After the Policies API is paginated in the API--which should come soon--we can use start passing the URL sort parameters that I have added here directly to the API call. That's why I decided to take the performance hit of making the API call for every change in sort, because it will make moving to backend sorting easier. 

## Checklist
- [ ] Investigated and inspected CI test results

### Here I tell how I validated my change

1. Inspect existing CI
2. Manual testing of each column sorted in each direction

Default with no sort specified in URL, sorts on Policy (name) column ascending (note the API response is not sort--sorting still happens on FE)
policies_with_no_sort_specified-uses_default_sort

Clicking on Policy column header adds sort parameters to the URL and reverses the sort on Policy column
policies_with_Policy_column_reversed

Clicking on Status column header sorts that column ascending
policies_with_Status_column_asc

Clicking on Status column header again reverses its sort
policies_with_Status_column_reversed

Clicking on Origin column header sorts that column ascending
policies_with_Origin_column_asc

Clicking on Origin column header again reverses its sort
policies_with_Origin_column_reversed

Clicking on Notifiers column header sorts that column ascending
policies_with_Notifiers_column_asc

Clicking on Notifiers column header again reverses its sort
policies_with_Notifiers_column_reversed

Clicking on Severity column header sorts that column ascending
policies_with_Severity_column_reversed

(note: I don't show clicking on Severity column header again to reverse it, but it works--it's just not an interesting use case, and I'm testing if the reviewer is paying attention :) )

Clicking on Lifecycle column header sorts that column ascending
policies_with_Lifecycle_column_asc

Clicking on Lifecycle column header again reverses its sort
policies_with_Lifecycle_column_reversed

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: ui/apps/platform/src/Containers/Policies/Table/PoliciesTablePage.tsx [hunk 1]
```diff
@@ -167,7 +193,8 @@ function PoliciesTablePage({
 
     useEffect(() => {
         fetchPolicies(query);
-    }, [query]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [query, sortOption]);
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Policies/Table/PoliciesTablePage.tsx:197]: Nit: you could update the call to `fetchPolicies` to require the sortOption as well, which would keep it consistent with the usage of `query` and allow us to remove this eslint-disable-next-line.

If I remember correctly, the object for `sortOption` is referentially equal while the options are unchanged.

File: ui/apps/platform/src/Containers/Policies/Table/PoliciesTable.tsx [hunk 2]
```diff
@@ -141,13 +99,10 @@ function PoliciesTable({
     const [enableDisableType, setEnableDisableType] = useState<EnableDisableType | null>(null);
 
     // index of the currently active column
-    const [activeSortIndex, setActiveSortIndex] = useState(0);
     // sort direction of the currently active column
-    const [activeSortDirection, setActiveSortDirection] = useState<SortDirection>('asc');
     // Handle Bulk Actions dropdown state.
     const [isActionsOpen, setIsActionsOpen] = useState(false);
     // For sorting data client side
-    const [rows, setRows] = useState<ListPolicy[]>([]);
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Policies/Table/PoliciesTable.tsx:150]: Nit: can we delete the associated comments here too?

**Comment by vjwilson** [comment]
> ...
> 
> I might lose sleep though wondering if the severity column _really_ does sort ascending...

Alright, here is proof

Screenshot 2024-03-12 at 10 12 37 AM

**Review by dvail — APPROVED** [review]
One thing I didn't catch on the initial review is that this change make the first sort of a column in the Policies table _descending_, instead of ascending.

To me this is arguably better behavior, at least for Severity, and matches the way the rest of the tables in the app behave. I think we should change the failing test to match the new behavior.

**Pull request merged**